chain/rust/src/lib.rs

```diff
@@ -122,7 +122,7 @@ pub type DeltaCoin = Int;
 
 pub type GenesisHashList = Vec<GenesisHash>;
 
-pub type NetworkId = u8;
+pub type NetworkId = u32;// TODO: u8? or custom struct? (u8 can't be exposed to wasm)
```

Comment:
if we don't encode type size anywhere we could probably live with u32, what do you think?

Reply:
It could give an API for testing against/for specific networks. e.g. `NetworkId::mainnet()`. Not all u32 values (in fact most wouldn't) work here either. That's easily addressed in the deserialization code but for setting it forces everything to return an error where you pass in a `u32`.

Reply:
I'd play it simple and have what can be easily exposed into wasm. in any case if we provide a couple of notes that network id is usually 0 or 1 people will figure out how to work with this